cli/command/service/logs.go

```diff
@@ -102,7 +176,7 @@ func (lw *logWriter) Write(buf []byte) (int, error) {
 		return 0, fmt.Errorf("invalid context in log message: %v", string(buf))
 	}
 
-	taskName, nodeName, err := lw.parseContext(string(parts[contextIndex]))
+	nodeID, serviceID, taskID, err := lw.parseContext(string(parts[contextIndex]))
```

Comment:
If you define a struct with nodeID/serviceID/taskID fields for indexing the cache map, `parseContext` can return that struct.